src/engine/sparqlExpressions/GeoExpression.cpp

```diff
@@ -111,10 +111,51 @@ SparqlExpression::Ptr makeGeoRelationExpression(SparqlExpression::Ptr child1,
                                                            std::move(child2));
 }
 
+// _____________________________________________________________________________
+template <SpatialJoinType Relation>
+std::optional<GeoFunctionCall> getGeoRelationExpressionParameters(
+    const SparqlExpression& expr) {
+  // Is this `expr` a call to `geof:sf[Relation](?x, ?y)`?
+  auto geoRelExpr = dynamic_cast<const GeoRelationExpression<Relation>*>(&expr);
+  if (geoRelExpr == nullptr) {
+    return std::nullopt;
+  }
+
+  // Extract variables
+  auto p1 = geoRelExpr->children()[0]->getVariableOrNullopt();
+  if (!p1.has_value()) {
+    return std::nullopt;
+  }
+  auto p2 = geoRelExpr->children()[1]->getVariableOrNullopt();
+  if (!p2.has_value()) {
+    return std::nullopt;
+  }
+
+  return GeoFunctionCall{Relation, p1.value(), p2.value()};
+}
+
 // _____________________________________________________________________________
 std::optional<GeoFunctionCall> getGeoFunctionExpressionParameters(
-    const SparqlExpression&) {
-  // TODO<ullingerc> handle geo relation functions in subsequent PR
+    const SparqlExpression& expr) {
+  // Check against all possible geo relation types
+  std::optional<GeoFunctionCall> res;
+  using enum SpatialJoinType;
+
+  if ((res = getGeoRelationExpressionParameters<INTERSECTS>(expr))) {
+    return res;
```

Comment:
This "inverse enum" case is somewhat vulnerable to adding a value to the enum and forgetting a case here. I think there is not a good way to do this (without C++26 reflection), but add a note to the definition of the enum + add a TODO<C++26 reflection> here to automatically get all the members of the enum.